### PR TITLE
Feat: Login

### PR DESCRIPTION
**Pull request by PedroChaparro**
**Note:** If you make some change in the .env file, stop the react-native server and run the `yarn start:cache` command to re-start the react-native server without caching the .env file.

Includes: 

1. Reusable button component. 
2. Login form UI. 
3. Login form validation . 
4. Auth context and useAuth hook to share the user session with any component. 
5. Save / recover the tokens from the async storage. 
6. Refresh the access token when the API respond with a 401 status code. 
7. Toasts alerts dependency and hook to make it reusable without repeating code. 

Preview:

https://user-images.githubusercontent.com/62714297/219883517-603a95b5-5767-4fb4-a192-97c0ab0f3485.mp4

### CODE REVIEW

**Review by Woynert — CHANGES_REQUESTED** [review]
Things I have experienced on my end. Running on an physical Android 10 device:
- The text color for the login fields are white. It has low contrast with the background which is gray-ish.
image

- There must not be password syntax check in the login screen, it is either a wrong password or a right password, no hints. 

- When you submit wrong credentials, currently the toast says "You are not logged in". I think it'd be better to instead show something like "Wrong email or password".

**Comment by PedroChaparro** [comment]
>     * There must not be password syntax check in the login screen, it is either a wrong password or a right password, no hints.

Without hints a new user will never know why their password is not valid.

**Comment by PedroChaparro** [comment]
> Things I have experienced on my end. Running on an physical Android 10 device...

Working on it.

**Comment by Woynert** [comment]
@PedroChaparro If the user forgot their password they must replace it. Those hints only help account stealers.

**Comment by PedroChaparro** [comment]
> @PedroChaparro If the user forgot their password they must replace it. Those hints only help account stealers.

Agree, I forgot this isn't the signup screen.

**Comment by PedroChaparro** [comment]
> Things I have experienced on my end. Running on an physical Android 10 device...

@Woynert, done

**Review by SilviaPabon — APPROVED** [review]
Great 🥳!

**Review by Woynert — CHANGES_REQUESTED** [review]
Tested on physical Android 10:
- The placeholder still has visibility issues
image

- Another issue I found is the login endpoint shouldn't treat mail as case sensitive, [I opened and issue for that](https://github.com/PedroChaparro/loomies-backend/issues/52).
- It seems that the first character you type in the mail fiel is upper case by default so the case sensitive issue is even more noticeable.

**Comment by PedroChaparro** [comment]
> The placeholder still has visibility issues

I think it's done this time. 

> Another issue I found is the login endpoint shouldn't treat mail as case sensitive, [I opened and issue for that](https://github.com/PedroChaparro/loomies-backend/issues/52).

I opened [a Pull Request](https://github.com/PedroChaparro/loomies-backend/pull/53) to close the issue.

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# On local environments this should be your computer local IP address
+API_URL = API IP or DNS url
```
Comment (Woynert) [inline .env.example:2]: I suggest adding an example like `http://127.0.0.1:8080` so that you don't have to guess how it will be interpreted.

**Review by Woynert — CHANGES_REQUESTED** [review]
About the `yarn start:cache` command please add it as foot note in the readme.

**Review by Woynert — APPROVED** [review]

**Pull request merged**